zk/conn.go

```diff
@@ -353,10 +423,12 @@ func (c *Conn) loop() {
 				if err == nil {
 					panic("zk: recvLoop should never return nil error")
 				}
-				close(closeChan) // tell send loop to exit
+				close(c.closeChan) // tell send loop to exit
 				wg.Done()
 			}()
 
+			c.resendZkAuth(reauthChan)
```

Comment:
I wanted to be explicit and make it more future-proof in case somebody someday decides to change this piece of code.